src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -665,19 +664,25 @@ static expected::ExpectedBytes TryASN1EncodeMenderCustomBinaryECFormat(
 	r.release();
 	s.release();
 
-	/* Allocate some array guaranteed to hold the DER-encoded structure */
-	vector<uint8_t> der_encoded_byte_array(mender_decode_buf_size);
-	unsigned char *arr_p = &der_encoded_byte_array[0];
-	int len = i2d_ECDSA_SIG(ecSig.get(), &arr_p);
+	/* Get the expected length in bytes of the DER encoded signature */
+	int len = i2d_ECDSA_SIG(ecSig.get(), NULL);
 	if (len < 0) {
 		return expected::unexpected(MakeError(
 			SetupError,
 			"Failed to set the signature parts in the ECDSA structure: "
 				+ GetOpenSSLErrorMessage()));
 	}
-	/* Resize to the actual size of the DER-encoded signature */
-	der_encoded_byte_array.resize(len);
 
+	/* Allocate an array guaranteed to hold the DER-encoded structure */
+	vector<uint8_t> der_encoded_byte_array(len);
+	unsigned char *arr_p = &der_encoded_byte_array[0];
```

Comment:
You can do `.data()` to get the pointer to the data part. And we should use the same type, I think, so `unsigned char` even for the vector.

Reply:
I think `.data()` is cleaner than getting an address of the first element. They are guaranteed to be the same (I hope!), but it's more C++-ish.